src/hub/manual/core/mat.rs

```diff
@@ -83,43 +83,133 @@ impl Mat {
     }
 
     #[inline(always)]
-    pub(crate) fn _at<T: DataType>(&self, i0: i32) -> Result<&T> {
-        self.match_format::<T>().and_then(|_| self.ptr(i0).map(|x| unsafe { &*(x as *const _ as *const T) }))
+    pub(crate) fn _at<T: DataType>(&self, row: i32) -> Result<&T> {
+        self.match_format::<T>()?;
+        let size = self.size()?;
+        if row >= size.height {
+            return Err(Error::new(
+                core::StsOutOfRange,
+                format!("Given coordinate is out of bounds. \
+                         Mat has size {:?} but was given row: {}", size, row)));
+        }
+        unsafe { self.at_unchecked(row) }
+    }
+
+    /// Like `Mat::at()` but performs no bounds or type checks
+    ///
+    /// Note that this (and `Mat::at()`) differs from the behavior of OpenCV's
+    /// `_Tp& cv::Mat::at(int i0 = 0)`, in that it only looks up by row, and
+    /// does not look up by column in the case of single-row Mats, as OpenCV's
+    /// `Mat::at()` does.
+    /// (This upstream behavior is documented at https://docs.opencv.org/3.4/d3/d63/classcv_1_1Mat.html#aa5d20fc86d41d59e4d71ae93daee9726)
```

Comment:
When I was looking into which dimension opencv's 1d `at()` method uses, I discovered they say it actually will use cols in a 1-row matrix. Since we implement `at()` using opencv's `ptr()` methods, we don't replicate this behavior. I'd like to add this documentation to the real `Mat::at()` function but I don't see a way for us today to append to the generated rustdoc of auto-generated functions. Open to suggestions here!